src/test_payload_dataset_manager.cc

```diff
@@ -217,31 +276,48 @@ TEST_CASE("PayloadDatasetManager::GetPayload")
 
   (*model_parser->Inputs())[name] = payload_model_tensor;
 
-  const std::string input_data_json{R"(
-      {
-        "data": [
-          {
-            "payload": [{}]
-          }
-        ]
-      }
-      )"};
+  SUBCASE("valid payload")
+  {
+    const std::string input_data_json{R"(
+        {
+          "data": [
+            {
+              "payload": [{}]
+            }
+          ]
+        }
+        )"};
+
+    auto data_loader{std::make_shared<MockDataLoader>()};
 
-  auto data_loader{std::make_shared<MockDataLoader>()};
+    const auto error{data_loader->ReadDataFromJSON(
+        model_parser->Inputs(), model_parser->Outputs(), input_data_json)};
 
-  const auto error{data_loader->ReadDataFromJSON(
-      model_parser->Inputs(), model_parser->Outputs(), input_data_json)};
+    REQUIRE(error.IsOk());
 
-  REQUIRE(error.IsOk());
+    const PayloadDatasetManager payload_dataset_manager(
+        data_loader, model_parser);
```

Comment:
I think we can take this out of subcase?

Reply:
This subcase is testing the valid path through the public member method GetPayload(), so it is needed.

Reply:
This takes the data loader as input, which takes the custom JSON string for this particular subcase, so I think it's more sensible to keep it in the subcase.

Reply:
Sure, I only pointed it out since it seemed like we could make the subcase more minimal but I'm fine if you feel strongly about keeping it this way for now.